### DISCUSSION TITLE
Server Error 500 when proposing a change to a project (via web)

### DISCUSSION THREAD

**Discussion by charminULTRA** [community/community#23383] [New to GitHub]
<p>Hi - I’m receiving this generic 500 server error when trying to propose a change to a project. It’s a project I don’t have edit access to so I believe it would create a new project and then a pull request. Not sure what to do.</p>
<p>Was trying to edit this: <a href="https://github.com/meshtastic/Meshtastic/tree-save/master/website/docs/software/community/atak.md" rel="noopener nofollow ugc">https://github.com/meshtastic/Meshtastic/tree-save/master/website/docs/software/community/atak.md</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/5/5530e0778e21e9c016c9141e6cd36108fb1369be.jpeg" data-download-href="/uploads/short-url/c9DiN7vieQ9lQ07oYrMM7JuGJm6.jpeg?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29433007/181076674-72ff6593-b455-4086-8157-972d8b85b63e.jpeg" alt="image" data-base62-sha1="c9DiN7vieQ9lQ07oYrMM7JuGJm6" width="654" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/5/5530e0778e21e9c016c9141e6cd36108fb1369be_2_654x500.jpeg, https://aws1.discourse-cdn.com/github/original/3X/5/5/5530e0778e21e9c016c9141e6cd36108fb1369be.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/5/5530e0778e21e9c016c9141e6cd36108fb1369be.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/5/5530e0778e21e9c016c9141e6cd36108fb1369be_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">876×669 102 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="charminULTRA" data-post="1" data-topic="189828">
<div class="title">
<div class="quote-controls"></div>
 charminULTRA:</div>
<blockquote>
<p>Was trying to edit this: <a href="https://github.com/meshtastic/Meshtastic/tree-save/master/website/docs/software/community/atak.md" rel="noopener nofollow ugc">https://github.com/meshtastic/Meshtastic/tree-save/master/website/docs/software/community/atak.md</a></p>
</blockquote>
</aside>
<p>when I open that link (and being logged into my GH account) I get:</p>
<blockquote>
<h3>You need to fork this repository to propose changes.</h3>
<p>Sorry, you’re not able to edit this repository directly—you need to fork it and propose your changes from there instead.</p>
</blockquote>
<p>if you’re landing directly on the error 500 page then something is definitely not working right.</p>
<p>You might consider cleaning your browser cache, just in case there are some cookies problems, and then try again the operation (ensuring you’re logged into your GH account).</p>

**Reply by karomilo13** [reply]
I am new at github

**Comment by charminULTRA** [comment]
<p>OK, I forked it and created my own repo, and tried to make the change but had the same error. I tried it on Firefox, Chrome, and even on mobile, with the same results.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="charminULTRA" data-post="3" data-topic="189828">
<div class="title">
<div class="quote-controls"></div>
 charminULTRA:</div>
<blockquote>
<p>OK, I forked it and created my own repo, and tried to make the change but had the same error.</p>
</blockquote>
</aside>
<p>What are you clicking on, exactly, before the 500 error page?</p>
<p>Are you now trying to change the <code>atak.md</code> document via GitHub’s browser UI?</p>
<p>Just to be clear we’re on the same wavelength here, to edit that file you should now open it in your own fork:</p>
<ul>
<li><a href="https://github.com/charminULTRA/Meshtastic/blob/master/website/docs/software/community/atak.md" rel="noopener nofollow ugc">https://github.com/charminULTRA/Meshtastic/blob/9c53287c80c8fbbe55908a38799b504781f8ad3b/website/docs/software/community/atak.md</a></li>
</ul>
<p>The page at the above link should show correctly. You should now be able to click on the <strong>Edit</strong> button on the bar above the file (pencil icon), which should allow you directly edit the file in the browser.</p>

**Comment by tajmone** [comment]
<p>PS. If you’re intending to propose your changes to the original (<em>upstream</em>) repository, you should then first create a new branch on your repository fork, do your changes to the <code>atak.md</code> file there, and then open a pull request on the upstream repository.</p>
<p>If you edit the file directly on the <code>master</code> branch of your fork, you’ll end up with your fork being out-of-synch with the upstream repository (i.e. <em>diverging</em>) and then you won’t be able to update your fork to mirror the original (i.e. fetch and merge changes from it into your fork) in the future. Also, if your pull request ends up being processed after some new changes were applied to the upstream repo, you’ll be asked to update your fork and rebase your branch to bring it en par with the master branch on the upstream, but if you apply your changes directly to master you won’t be able to do so without loosing your changes.</p>

**Comment by charminULTRA** [comment]
<p>Right, I believe we are on the same wavelength. I used the link you provided, hit Edit, made my edit and selected “Create a new branch”. After hitting Commit Changes, it takes me to the error page.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/9/a99d3ad7b3e0974b170659ca49233c467505570d.png" data-download-href="/uploads/short-url/octysVURlHTL43ufhMwKTPaGIQl.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29433007/181076678-9c0a6ea1-c13e-4ad0-9f9a-b99cd86140be.png" alt="image" data-base62-sha1="octysVURlHTL43ufhMwKTPaGIQl" width="690" height="86" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/9/a99d3ad7b3e0974b170659ca49233c467505570d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">733×92 2.72 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="charminULTRA" data-post="6" data-topic="189828">
<div class="title">
<div class="quote-controls"></div>
 charminULTRA:</div>
<blockquote>
<p>Right, I believe we are on the same wavelength. I used the link you provided, hit Edit, made my edit and selected “Create a new branch”. After hitting Commit Changes, it takes me to the error page.</p>
</blockquote>
</aside>
<p>Ok, so this time you did manage to create a new branch, but only committing changes failed?</p>
<p>This error 500 page you’re getting is a rather generic error, so it doesn’t really say much about what’s going on. I’ve checked the GitHub Status page and it doesn’t report any  services being down right now:</p>
<p><a href="https://www.githubstatus.com/" class="onebox" target="_blank" rel="noopener nofollow ugc">https://www.githubstatus.com/</a></p>
<p>So the problem is not related to server problems. At this point, I suspect that something is not working properly on the client side, i.e. a problem with your browser or Internet connection.</p>
<p>I use the GitHub WebUI a lot, but not for Git operations though — I do all repository editing locally via Git, and only use the WebUI for pull requests, Issues, etc.</p>
<p>If the problem is due to internet connection failures, or some browser problems, etc., you end up needing to refresh the page (as mentioned in the error page), which puts you at risk of loosing your contents changes.</p>
<p>My advise would be to do one operation at the time, when working in the browser — i.e. first create the branch, then edit the file and commit changes, and then create a pull request. Being in control of each step is better if these errors are frequent.</p>
<p>Also, I suggest you do all editing work in an actual editor, and only when the changed file is ready you then click on Edit, paste your changed contents over the originals (select all and paste over), so that the operation doesn’t take too long — the problem might be related to timeout. At least, if you get the 500 error page, you don’t loose your changes, which are still in the editor.</p>
<p>I also strongly advice against navigating back in history during these delicate operations, since cookies play an important role in account related operations, and using the “back/forward” buttons in the browser is not the same as navigating naturally through GH WebUI, for they might revert some page-related status which GH needs to track your current user session. Also, having multiple GH pages at once might lead to problems, especially if more than on browser tab points at the same page.</p>

**Comment by tajmone** [comment]
<p>Also, check that you don’t have some browser extension that might interfere with the process (e.g. ads or pop-ups blockers) or some invasive anti-virus or anti-malware tool that monitors Internet navigation and (for example) prevents sending user data over the Internet, or interferes in some other way with your session account.</p>
<p>Basically, anything that could interfere with your browser session (anonymizing tools, proxies, whatever)</p>

**Comment by charminULTRA** [comment]
<p>I didn’t create a new branch, because that button says it does it for you, but it fails?</p>
<p>Doesn’t seem like a browser or internet issue, since I tried different browsers and devices, and even put my iPhone on LTE rather than Wifi to try different internet connections.</p>
<p>I’ll try creating a branch without using that button.</p>

**Comment by charminULTRA** [comment]
<p>Well creating a branch and making changes didn’t work either, same error :frowning:</p>

**Comment by charminULTRA** [comment]
<p>Clearing cookies did not change anything either.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="charminULTRA" data-post="10" data-topic="189828">
<div class="title">
<div class="quote-controls"></div>
 charminULTRA:</div>
<blockquote>
<p>Well creating a branch and making changes didn’t work either, same error</p>
</blockquote>
</aside>
<p>Then you might have to wait for some GH staff members to check what’s happening on their servers, maybe from their logs they’d be able to provide more details on what’s happening.</p>
<p>I do remember though that in the past years, during  the Summer heat, we did experience all sorts of Internet related problems, due to overheating of electronic devices. I myself had to replace a monitor that “fried” in front my eyes, and a few other devices that overheated and were damaged.</p>
<p>I’m just surprised that the error is always a generic one, usually server-side problems are able to provide specific details of what happens. So I tend to still think that the problem is either on your side (PC or router) or somewhere along the network line — either a time out, or data being corrupted in the process.</p>
<p>Why don’t you try doing the same operations using Git command line, or GitHub Desktop? This would allow to narrow down the problem (if it works, then there problem is either with the browser(s) or the WebUI).</p>

**Reply by MeryJan1** [reply]
Git komut satırını kullanırken de hata alıyorum.

**Comment by charminULTRA** [comment]
<p>Thanks for the continued help…I’ll give that a shot tomorrow. Interestingly, I tried creating a repo from scratch and was able to add files and edit them without issues.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="charminULTRA" data-post="13" data-topic="189828">
<div class="title">
<div class="quote-controls"></div>
 charminULTRA:</div>
<blockquote>
<p>Interestingly, I tried creating a repo from scratch and was able to add files and edit them without issues.</p>
</blockquote>
</aside>
<p>That’s strange indeed. These are repository operations too.</p>
<p>I checked the repository, to see if it contained code owner rules which might be blocking code changes, but it didn’t. I only noticed that it contains Git modules, but this should interfere with these operations.</p>
<p>Let me now how testing with Git/GitHub Desktop went!</p>

**Comment by charminULTRA** [comment, marked as answer]
<p>Pushing the change from GitHub Desktop was successful, with no issues, and I was able to view the edit on GitHub Web! I guess I’ll just go that route from here on out.</p>

**Comment by charminULTRA** [comment]
<p>Also - Thanks again for the help!</p>
